Detectors/PHOS/calib/PHOSCalibWorkflow/src/PHOSHGLGRatioCalibDevice.cxx

```diff
@@ -32,109 +29,53 @@ using namespace o2::phos;
 void PHOSHGLGRatioCalibDevice::init(o2::framework::InitContext& ic)
 {
 
-  mMapping.reset(new o2::phos::Mapping(""));
-  if (!mMapping) {
-    LOG(FATAL) << "Failed to initialize mapping";
-  }
-  if (mMapping->setMapping() != o2::phos::Mapping::kOK) {
-    LOG(ERROR) << "Failed to construct mapping";
-  }
-  mRawFitter.reset(new o2::phos::CaloRawFitter());
-
   //Create histograms for mean and RMS
   short n = o2::phos::Mapping::NCHANNELS;
   mhRatio.reset(new TH2F("HGLGRatio", "HGLGRatio", n, 0.5, n + 0.5, 100, 10., 20.));
+  mCalibParams.reset(new CalibParams());
 }
 
 void PHOSHGLGRatioCalibDevice::run(o2::framework::ProcessingContext& ctx)
 {
 
-  //
-  for (const auto& rawData : framework::InputRecordWalker(ctx.inputs())) {
-
-    o2::phos::RawReaderMemory rawreader(o2::framework::DataRefUtils::as<const char>(rawData));
-
-    // loop over all the DMA pages
-    while (rawreader.hasNext()) {
-      mMapPairs.clear();
+  // scan Cells stream, collect HG/LG pairs
+  if (mRunStartTime == 0) {
+    mRunStartTime = o2::header::get<o2::framework::DataProcessingHeader*>(ctx.inputs().get("cellTriggerRecords").header)->startTime;
+  }
 
-      try {
-        rawreader.next();
-      } catch (RawDecodingError::ErrorType_t e) {
-        LOG(ERROR) << "Raw decoding error " << (int)e;
-        //if problem in header, abandon this page
-        if (e == RawDecodingError::ErrorType_t::PAGE_NOTFOUND ||
-            e == RawDecodingError::ErrorType_t::HEADER_DECODING ||
-            e == RawDecodingError::ErrorType_t::HEADER_INVALID) {
-          break;
+  auto cells = ctx.inputs().get<std::vector<o2::phos::Cell>>("cells");
```

Comment:
why vector and not span?

Reply:
rudiment: tested reason of seg. violation and forgot to return back